std/assembly/array.ts

```diff
@@ -369,19 +371,27 @@ export class Array<T> {
     return sliced;
   }
 
-  splice(start: i32, deleteCount: i32 = i32.MAX_VALUE): void {
-    if (deleteCount < 1) return;
+  splice(start: i32, deleteCount: i32 = i32.MAX_VALUE): Array<T> {
+    if (deleteCount < 1) return new Array<T>();
```

Comment:
Is this check necessary? Feels like the function should also work properly without it.

Reply:
done

Reply:
I revert back this. Without got unreachables inside memory.move I guess

Reply:
I could remove `if (deleteCount < 1)` but this required modify this:
```ts
deleteCount = min(deleteCount, length - start);
```
to this:
```ts
deleteCount = deleteCount <= 0 ? 0 : min(deleteCount, length - start);
```
Is it ok?

Reply:
Can remove the branch with `max(min(deleteCount, length - start), 0)`

Reply:
I tried but this produce panic about out of bound access somewhere

Reply:
Ok, it seems just needed specialize `max/min`. All issues fixed

Reply:
Hmm, regarding `max<i32>(min<i32>(deleteCount, length - start), 0)` it appears that all variables involved are declared `i32`s. And this doesn't work without type arguments?

Reply:
yep

Reply:
I think `length` infers like `usize` and all `max/min` resolves as `min<usize>/max<usize>` without explicitly specialized